central/role/datastore/datastore_impl.go

```diff
@@ -152,6 +152,27 @@ func (ds *dataStoreImpl) GetAllRoles(ctx context.Context) ([]*storage.Role, erro
 	return ds.getAllRolesNoScopeCheck(ctx)
 }
 
+func (ds *dataStoreImpl) GetRolesFiltered(ctx context.Context, filter func(role *storage.Role) bool) ([]*storage.Role, error) {
+	if ok, err := roleSAC.ReadAllowed(ctx); !ok || err != nil {
```

Comment:
Why `!ok` results in no error here?

Reply:
I guess I just copied it from the other role implementations.
There seems to be a mix within the datastore - some use  the construct above, others use `sac.VerifyAuthZOK` (e.g. upserts, remove). While it's a change to the API, it feels more natural to me. I'll move the whole store to use `sac.VerifyAuthZOK`.